core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -109,6 +110,29 @@ public RestStatus status() {
         return getShardInfo().status();
     }
 
+    /**
+     * Gets the location of the written document as a string suitable for a {@code Location} header.
+     * @param routing any routing used in the request. If null the location doesn't include routing information.
+     */
+    public String getLocation(@Nullable String routing) {
```

Comment:
I don't feel *too* bad about them having to pass in the routing information here. I figure routing is kind of an advanced use case. And if they are using the transport client their IDE will force them to pass in some string and it isn't a huge leap to read the javadoc from there.... I get your point though. If it were easy to pass the routing along I wouldn't have made this option. I can have another look and see if I can get the routing though.

What do you think?